src/audio/dai-zephyr.c

```diff
@@ -1283,17 +1342,47 @@ int dai_zephyr_copy(struct dai_data *dd, struct comp_dev *dev)
 
 	buffer_release(buf_c);
 
-	buf_c = buffer_acquire(dd->local_buffer);
 
 	/* calculate minimum size to copy */
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
+		buf_c = buffer_acquire(dd->local_buffer);
 		src_samples = audio_stream_get_avail_samples(&buf_c->stream);
 		sink_samples = free_bytes / sampling;
 		samples = MIN(src_samples, sink_samples);
+		buffer_release(buf_c);
 	} else {
+		struct list_item *sink_list;
+
+		/*
+		 * In the case of capture DAI's with multiple sinks, compute the minimum number
```

Comment:
Took me some time to check what you mean by a "sink" here - looks like an output, right? I think we call opposite things by a "sink pin" and a "sink buffer," when you say "a DAI with multiple sinks" I imaging sink pins, i.e. inputs, which isn't the case here?

Reply:
right I mean sink buffers. Will modify the wording to say sink buffers.